### PR TITLE
Fix Multiple Build Errors on different platforms

### PR DESCRIPTION
**Pull request by trcrsired**
Particularly for canadian compilation

### CODE REVIEW

File: llvm/include/llvm/TargetParser/Triple.h [hunk 1]
```diff
@@ -46,64 +46,64 @@ class Triple {
   enum ArchType {
     UnknownArch,
 
-    arm,            // ARM (little endian): arm, armv.*, xscale
```
Comment (asl) [inline llvm/include/llvm/TargetParser/Triple.h:49]: Please do not do such large code reformatting. It makes impossible to review them

File: lldb/source/Host/CMakeLists.txt [hunk 2]
```diff
@@ -108,13 +108,9 @@ else()
       linux/HostInfoLinux.cpp
       linux/LibcGlue.cpp
       linux/Support.cpp
+      android/HostInfoAndroid.cpp
+      android/LibcGlue.cpp
       )
-    if (CMAKE_SYSTEM_NAME MATCHES "Android")
```
Comment (asl) [inline lldb/source/Host/CMakeLists.txt:112]: What is wrong with android detection? I doubt this is a proper solution

Comment (trcrsired) [reply lldb/source/Host/CMakeLists.txt:112]: I use "Linux" to do canadian compilation from linux to android without using android NDK. That is why this is not correct.

File: llvm/include/llvm/TargetParser/Triple.h (hunk 1 above)
Comment (trcrsired) [reply llvm/include/llvm/TargetParser/Triple.h:49]: without doing code formatting the CI won't pass.

Comment (asl) [reply llvm/include/llvm/TargetParser/Triple.h:49]: You can ignore these.

File: lldb/source/Host/CMakeLists.txt (hunk 2 above)
Comment (asl) [reply lldb/source/Host/CMakeLists.txt:112]: Well, so this is not a cmake problem then.

File: llvm/include/llvm/TargetParser/Triple.h (hunk 1 above)
Comment (trcrsired) [reply llvm/include/llvm/TargetParser/Triple.h:49]: i will fix it later. but first pass CI with other things first to ensure it does not break anything.

File: lldb/source/Host/CMakeLists.txt (hunk 2 above)
Comment (trcrsired) [reply lldb/source/Host/CMakeLists.txt:112]: > Well, so this is not a cmake problem then.

but it still needs a solution tbh. android is linux.

Comment (asl) [reply lldb/source/Host/CMakeLists.txt:112]: Not quite. It's distant enough in many aspects (from the toolchain perspective) to have a separate cmake system name.

Comment (trcrsired) [reply lldb/source/Host/CMakeLists.txt:112]: That is probably because cmake treats Android differently, despite it can be treated the same.

Comment (trcrsired) [reply lldb/source/Host/CMakeLists.txt:112]: I will try to find a better solution.

Comment (JDevlieghere) [reply lldb/source/Host/CMakeLists.txt:112]: While Android might be Linux, the inverse isn't true. Amongst other things, this change would result in all Linux hosts reporting that they support Android triples.

**Review by JDevlieghere — CHANGES_REQUESTED** [review]

**Pull request closed without merging**